test/testsuite.jl

```diff
@@ -17,6 +17,12 @@ using Adapt
 struct ArrayAdaptor{AT} end
 Adapt.adapt_storage(::ArrayAdaptor{AT}, xs::AbstractArray) where {AT} = AT(xs)
 
+≈(a, b) = Base.isapprox(a, b)
```

Comment:
Maybe we should just use a separate function not named \approx and move the other special-case (`cpu_out isa Tuple && gpu_out isa Tuple`) to a dispatch rule as well.